libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -304,7 +310,20 @@ private async Task<PromptRecognizerResult<TokenResponse>> RecognizeTokenAsync(IT
             }
             else if (IsTeamsVerificationInvoke(context))
             {
-                // TODO: add missing code
+                var magicCodeObject = context.Activity.Value as JObject;
+                var magicCode = magicCodeObject.GetValue("state").ToString();

```

Comment:
what if GetValue("state") return null
using "as string" rather than "ToString()" allows a null to be checked later...

Reply:
No error will be returned, the user will just not get a token.  This already happens when the prompt is initialized i.e. before they enter the code.  This will not blow up.  It is ok to call the ` return await adapter.GetUserTokenAsync(context, _settings.ConnectionName, magicCode, default(CancellationToken)).ConfigureAwait(false);` method with a null magic code.  This will not throw an error and will not blow up.